File: CPP/Program/Kategori.cpp
#include <iostream>

using namespace std;

class Kategori{
    // atribut private
    private:
        string id_kategori; // id kategori (string) agar bisa kombinasi
        string nama_kategori;
        string deskripsi;

    public:
        // constructor kosong
        Kategori(){}

        // constructor
        Kategori(string id_kategori, string nama_kategori, string deskripsi){
            this->id_kategori = id_kategori;
            this->nama_kategori = nama_kategori;
            this->deskripsi = deskripsi;
        }

        // getter
        string getIDKategori() const {
            return id_kategori;
        }
        string getNamaKategori() const {
            return nama_kategori;
        }
        string getDeskripsi() const {
            return deskripsi;
        }

        // setter
        void setIDKategori(string id_kategori) {
            this->id_kategori = id_kategori;
        }
        void setNamaKategori(string nama_kategori) {
            this->nama_kategori = nama_kategori;
        }
        void setDeskripsi(string deskripsi) {
            this->deskripsi = deskripsi;
        }

        // destructor
        ~Kategori() { }
};

File: Python/Program/Kategori.py
class Kategori:
    def __init__(self, id_kategori="", nama_kategori="", deskripsi=""):
        self.__id_kategori = id_kategori
        self.__nama_kategori = nama_kategori
        self.__deskripsi = deskripsi

    # getter
    def get_id_kategori(self):
        return self.__id_kategori
    def get_nama_kategori(self):
        return self.__nama_kategori
    def get_deskripsi(self):
        return self.__deskripsi

    # setter
    def set_id_kategori(self, id_kategori):
        self.__id_kategori = id_kategori
    def set_nama_kategori(self, nama_kategori):
        self.__nama_kategori = nama_kategori
    def set_deskripsi(self, deskripsi):
        self.__deskripsi = deskripsi
